### PR TITLE
Use deferred start-epc for #193

### PR DESCRIPTION
**Pull request by yuutayamada**

### CODE REVIEW

File: nim-suggest.el [hunk 1]
```diff
@@ -55,6 +55,10 @@
     :prefix) ; matching state. See also prefixmatches.nim
   "Keywords for SexpNode type on nimsuggest.nim.")
 
+(defvar nimsuggest-accept-process-timeout 150)
+(defvar nimsuggest-accept-process-timeout-count 100)
```
Comment (krux02) [inline nim-suggest.el]: Can you add a documenation string for these variables? For me it's not clear what unit the timeout is, and what the count is actually counting.

**Comment by krux02** [comment]
When you are about to add threshold values, I think it is a good idea to add a threshold for maximum memory usage as well. Last time I used nimsuggest after an hour or two the nimsuggest process used up so much memory that my computer started swapping. That is something I would like to be able to prevent. Nimsuggest just needs to be restarted in such a case.

File: nim-suggest.el (hunk 1 above)
Comment (yuutayamada) [reply nim-suggest.el]: will do. It's just timeout values how long you can wait if nimsuggest doesn't answer for its port for first connection.

Comment (yuutayamada) [reply nim-suggest.el]: I meant until emacs side gets port number for nimsuggest server. In current implementation, if the timeout happen, it just block the query to nimsuggest for the buffer, so annoying frozen should not occur again in the buffer. I hope.

Comment (krux02) [reply nim-suggest.el]: I think you shouldn't block emacs for 150ms. Just assume that the start of nimsuggest needs to be async. On a 60Hz monitor 150ms is 9 frames of emacs hanging there doing nothing.

Comment (yuutayamada) [reply nim-suggest.el]: I think you are misunderstanding. the `deferred-wait` doesn't stop Emacs IIUC

**Pull request merged**